index.html

```diff
@@ -11,32 +11,201 @@
+				padding: 1em;
+				border: 2px solid rgba(0, 0, 0, .5);
+			}
+
+			#search-field, #search-button {
+				height: 2em;
+				font-size: 18px;
+			}
+
+			#search-button {
+				margin-left: 1em;
+				display: inline-block;
+			}
+
 		</style>
 	</head>
 
 	<body>
-		<h1>Thrilling Frontend Technical Challenge</h1>
+		<div class="container">
+			<h1 class="page-title">Thrilling Frontend Technical Challenge</h1>
+
+			<form class="search-form">
+				<input type="text" placeholder="Enter Search Term" id="search-field" />
+				<input type="button" value="Search" id="search-button" />
+			</form>
 
-		<h2>Open your Developer Tools Console to view output.</h2>
+			<div id="results-list"></div>
+		</div>
 
 		<script>
-			var request = new XMLHttpRequest();
 
-			request.open('GET', 'https://api.nytimes.com/svc/topstories/v2/science.json?api-key=Gwxln5M3geWlhR6UE0TY1FUWKSG3wCil');
+			const resultsList = document.querySelector('#results-list');
+			const searchField = document.querySelector('#search-field');
+			const searchButton = document.querySelector('#search-button');
+
+			searchField.addEventListener('keydown', (event) => {
+				if (event.keyCode === 13) {
+					event.preventDefault();
+					const request = setupRequest();
+					request.send();
+				}
+			});
+
+			searchButton.addEventListener('click', (event) => {
+				event.preventDefault();
+
+				const request = setupRequest();
+				request.send();
+			});
+
+
+			function setupRequest () {
+				const request = new XMLHttpRequest();
+
+				request.open('GET', 'https://api.nytimes.com/svc/topstories/v2/science.json?api-key=Gwxln5M3geWlhR6UE0TY1FUWKSG3wCil');
+
+				request.onload = handleResponse;
+
+				request.onerror = function() {
+					console.error(this);
+				};
+
+				return request;
+			}
+
+			function handleResponse () {
 
-			request.onload = function() {
 				if (this.status >= 200 && this.status < 400) {
-					console.log(JSON.parse(this.response));
+					const parsedResponse = JSON.parse(this.response);
+					if (parsedResponse.results && Array.isArray(parsedResponse.results)) {
+
+						// remove existing children from results
+						while (resultsList.firstChild) {
+							resultsList.removeChild(resultsList.firstChild);
+						}
+
+						// split search input into array so order doesn't matter
+						const searchInput = searchField.value.split(' ');
+						let matchedResults = [];
+
+						parsedResponse.results.forEach((item) => {
+							// combine all searchable fields into one block of text
+							const searchableText = `${item.title} ${item.byline} ${item.section}`.toLowerCase();
+							let match = false;
+							let relevance = 0;
+
+							// ignore blank spaces in search input
+							if (searchInput.length > 0 && searchInput[0] !== '') {
+								searchInput.forEach((searchWord) => {
+
+									// basic regex search pattern so we match parts of words as well
+									// e.g. chan would work for change
+									const pattern = new RegExp(searchWord, 'g');
```

Comment:
```suggestion
									const pattern = new RegExp(searchWord, 'gi');
```
Consider adding case insensitive flag to regex to more generally solve case insensitive searching.

Reply:
That's really helpful, admittedly regex are not a strong point for me. I need to use them more often and learn more.

Reply:
it's crazy how powerful regex can be when you get into capturing groups and look aheads and behinds. but then I wonder if it's really a good idea to dive too deeply into, since it creates code that looks like total gibberish and that other developers would have a hard time deciphering. if it were any other type of syntax cleverness it would never make it through a code review, yet the small tricks can be very helpful. I especially like it for doing powerful find and replace across multiple files in my text editor.

anyway, I'm a big fan of https://regex101.com/ as a tool for testing/learning